### DISCUSSION TITLE
.gitignore isn't doing what it's supposed to be doing

### DISCUSSION THREAD

**Discussion by yungkevster** [community/community#32350] [Actions]
I am new to git and I want to choose which files I want to push. Some files are private for private purposes obviously so I dont want git to track them. I created a .gitignore file and I added which folder and files I dont want to be tracked. So in this case I dont want the Classes folder to be tracked and I dont want the settings.json file to be tracked either. So I saved the .gitignore file and did

git add .
git commit -m "test"
git push origin master

but every single file still got pushed into my rep on github

Also as you could see the Class folder is still connected to my git rep because everytime I do a change for instance the file in the folder turns orange as you could see on main.cpp

![image](https://user-images.githubusercontent.com/113138995/189229156-61ceaa39-fc28-4672-8e17-a500144ab8e2.png)

**Comment by airtower-luna** [comment]
The `.gitignore` file [affects only untracked files](https://git-scm.com/docs/gitignore#_description). If files are already committed they won't disappear from the repository just because they're listed in the ignore file. If you want to stop tracking those files, you need to remove them from the repository, you can use [`git rm --cached`](https://git-scm.com/docs/git-rm#Documentation/git-rm.txt---cached) to remove a file from the repository without removing it from your working tree. Note that they'll still be part of the repository history, unless you rewrite the history to delete them.

In general, I recommend you avoid `git add .`, and add specific files instead. Much easier to avoid committing things by mistake that way. :wink:

**Reply by yungkevster** [reply]
Ok thank you for your response! By the way are the pathways I listed in my .gitignore file correct? I want to remove tracking of the folder Classes and the file settings.json

This is only a test rep so I'm basically going to redo it and do git add .gitignore before pushing any other files

**Reply by yungkevster** [reply]
> The `.gitignore` file [affects only untracked files](https://git-scm.com/docs/gitignore#_description). If files are already committed they won't disappear from the repository just because they're listed in the ignore file. If you want to stop tracking those files, you need to remove them from the repository, you can use [`git rm --cached`](https://git-scm.com/docs/git-rm#Documentation/git-rm.txt---cached) to remove a file from the repository without removing it from your working tree. Note that they'll still be part of the repository history, unless you rewrite the history to delete them.
> 
> In general, I recommend you avoid `git add .`, and add specific files instead. Much easier to avoid committing things by mistake that way. 😉

So it's still not working. I decided to delete my rep and start fresh so what I did is:

Created a new rep on github
In VSCODE: I typed:

```
git init
git add .gitignore
git commit -m "first commit"
git remote add origin https://github.com/yungkevster/Test-Rep.git
git push origin master
```

So now I only added .gitignore and I wanted to try it out by seeing if `git add .` would add all the files or ignore the ones i stated that I want ignored so I typed:

```
git add .
git commit -m "test"
git push origin master

```
And unfortunately every single folder and file was pushed to my repo :/

How do I do this?

**Reply by airtower-luna** [reply]
Do you actually have a `C++/` directory *within* your repository? The screenshot looks a bit like `C++/` is the name of the repository, not a path within it.

**Reply by yungkevster** [reply]
![image](https://user-images.githubusercontent.com/113138995/189234858-8527f558-dc45-49c2-8429-a3aec845e908.png)

Yea the .git is inside the C++ folder as shown in the image

**Reply by airtower-luna** [reply]
Then you need to remove the `C++` part from the `.gitignore` entries. Something like
```
Classes/
.vscode/settings.json
```
should work.

**Reply by yungkevster** [reply]
Ok let me try :)

**Reply by yungkevster** [reply]
> Then you need to remove the `C++` part from the `.gitignore` entries. Something like
> 
> ```
> Classes/
> .vscode/settings.json
> ```
> 
> should work.

I see now it looks like it's going to work. I only did git init
and all the files I wanted to be ignored are gray

![image](https://user-images.githubusercontent.com/113138995/189237690-08c8a188-d668-4028-9117-4f15d0a5f12b.png)
